test/apps/echoer/main.go

```diff
@@ -15,12 +15,15 @@ func main() {
 	port := os.Getenv("PORT")
 	addr := ":" + port
 
-	if _, err := discoverd.AddServiceAndRegister(service, addr); err != nil {
+	hb, err := discoverd.AddServiceAndRegister(service, addr)
+	if err != nil {
 		log.Fatal(err)
 	}
+	defer hb.Close()
```

Comment:
In one of the commit messages: "Use the pkg/shutdown system in all flynn services.  Small test/demo/example apps just use plain defers and idiomatic go error handling, to make it perfectly clear there's nothing special about the flynn shutdown convenience package."